examples/[2] Constraints.ipynb

```diff
@@ -0,0 +1,508 @@
+       " ('.', '.')]"
+      ]
+     },
+     "execution_count": 37,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "sentence = 'Jack Black starred in the 2003 film classic \"School of Rock\".'\n",
+    "get_entities(sentence)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We flattened the tree of entities, so the return format is a list of `(word, entity type)` tuples. Looks like we have three labels here: 'Jack' and 'Black' labeled as 'PERSON', 'School' as a proper noun, and 'Rock' as a 'GPE' (geo-political entitity: it thinks Rock is the name of a place, a city or something). Whatever technique NLTK uses for named entity recognition may be a bit rough, but it did a pretty decent job here!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating our NamedEntityConstraint\n",
+    "\n",
+    "Now that we know how to detect named entities using NLTK, let's create our custom constraint."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from textattack.constraints import Constraint\n",
+    "\n",
+    "class NamedEntityConstraint(Constraint):\n",
+    "    \"\"\" A constraint that ensures `x_adv` only substitutes named entities from `x` with other named entities.\n",
+    "    \"\"\"\n",
+    "    def __call__(self, x, x_adv, original_text=None):\n",
+    "        x_entities = get_entities(x.text)\n",
+    "        x_adv_entities = get_entities(x_adv.text)\n",
+    "        # If there aren't named entities, let's return False (the attack\n",
+    "        # will eventually fail).\n",
+    "        if len(x_entities) == 0:\n",
+    "            return False\n",
+    "        if len(x_entities) != len(x_adv_entities):\n",
+    "            # If the two sentences have a different number of entities, then \n",
+    "            # they definitely don't have the same labels. In this case, the \n",
+    "            # constraint is violated, and we return True.\n",
+    "            return False\n",
+    "        else:\n",
+    "            # Here we compare all of the words, in order, to make sure that they match.\n",
+    "            # If we find two words that don't match, this means a word was swapped \n",
+    "            # between `x` and `x_adv`. That word must be a named entity to fulfill our\n",
+    "            # constraint.\n",
+    "            x_word_label = None\n",
+    "            x_adv_word_label = None\n",
+    "            for (word_1, label_1), (word_2, label_2) in zip(x_entities, x_adv_entities):\n",
+    "                if word_1 != word_2:\n",
+    "                    # Finally, make sure that words swapped between `x` and `x_adv` are named entities. If \n",
+    "                    # they're not, then we also return False.\n",
+    "                    if (label_1 not in ['NNP', 'NE']):\n",
```

Comment:
Good catch!! I think this is just because I put a checkpoint in the PR that didn't have the `or` in the if statement on line 263. I re-saved and it's in now.